handlers/telegram/telegram_test.go

```diff
@@ -613,7 +677,23 @@ var testCases = []ChannelHandleTestCase{
 		URL:                  "/c/tg/8eb23e93-5ecb-45ba-b726-3b064e0c568c/receive/",
 		Data:                 noOkFile,
 		ExpectedRespStatus:   200,
-		ExpectedBodyContains: "no 'ok' in response",
+		ExpectedBodyContains: "not present",
+	},
+	{
+		Label:                "Receive invalid JSON File response",
+		URL:                  "/c/tg/8eb23e93-5ecb-45ba-b726-3b064e0c568c/receive/",
+		Data:                 invalidJsonFile,
+		ExpectedRespStatus:   200,
+		ExpectedBodyContains: "unable to resolve file",
+		ExpectedErrors:       []*courier.ChannelError{courier.ErrorResponseUnparseable("JSON")},
+	},
+	{
+		Label:                "Receive error File response",
+		URL:                  "/c/tg/8eb23e93-5ecb-45ba-b726-3b064e0c568c/receive/",
+		Data:                 errorFile,
+		ExpectedRespStatus:   200,
+		ExpectedBodyContains: "unable to resolve file",
+		ExpectedErrors:       []*courier.ChannelError{courier.ErrorResponseUnparseable("JSON")},
```

Comment:
isn't this test case supposed to test catching an error code?

Reply:
we are parsing the response into struct now so if 'ok' is not present the struct will have that as false so we catch both the same way

Reply:
the JSON error we catch now are new tests I added